src/asynciterable/operators/_flatten.ts

```diff
@@ -93,10 +93,12 @@ export class FlattenConcurrentAsyncIterable<TSource, TResult> extends AsyncItera
               }
               if (active < concurrent) {
                 pullNextOuter(value as TSource);
+                results[0] = outer.next();
               } else {
+                // remove the outer iterator from the race, we're full
+                results[0] = NEVER_PROMISE;
                 outerValues.push(value as TSource);
               }
-              results[0] = outer.next();
```

Comment:
Thanks for calling this out.

This operator is technically working as designed, i.e. we are intentionally pulling and buffer values from the outer sequence as quickly as they're emitted, but only flattening `concurrent` number of inner sequences at a time (similar to the Observable implementation).

But I can see how the internal buffering is problematic, and I agree that we shouldn't need to buffer the outer sequence values.

Reply:
Thanks @trxcllnt, I manged to get the tests running locally, thanks for your help identifying the problems with my code.

When I started looking at the actual code I figured it must have been by design.

My intent was to fan out to do multiple API requests ( to AWS SQS) concurrently based on the result of of a dynamodb scan and the speed that I could processes the items is reduced significantly when I removed the concurrency and tried a `concatMap` ( although this technically works, and solves my memory issues, although a little slower for my application).

My initial motivation for moving from RX to IX was to fix this issue, figuring it would be difficult to prevent my memory issues unless I had a way of throttling the producer, I assume another approach that would work as the library currently stands is to use a `throttle` before the `flatMap` and attempt to tweak that time so the consumer can keep up, and the buffer never explodes.